networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1935,3 +1935,88 @@ def dist_path(v):
+    some blocked edges.
+
+    CTP solved using a repositioning algorithm.
+    First, the initial path is found using Dijkstra's algorithm and then the
+    repostitioning algorithm is implemented using taking into consideration
+    the edges that are blocked.
+
+    Parameters
+    ----------
+    G: A weighted graph.
+
+    blocked: list
+        A list with the blocked edges.
+
+    starting: node
+        Starting node.
+
+    ending: node
+        Ending node.
+
+    weight: string, optional (default='weight')
+        Represents the edge weight of each edge data key.
+
+    Returns
+    -------
+    dis: list
+        The path followed taking into consideration the blocked edges.
+
+    Example
+    -------
+    >>> import networkx as nx
+    >>> e = [(0, 1, 4), (0, 2, 6), (1, 2, 1), (1, 3, 8), (3, 2, 1), (3, 4, 16),
+    ...        (4, 2, 1), (4, 5, 32), (5, 2, 1)]
+    >>> G = nx.MultiGraph()
+    >>> G.add_weighted_edges_from(e)
+    >>> blocked = [(1,2), (3,2), (4,2)]
+    >>> output = ctp(G, blocked, 0, 2)
+    >>> print(output)
+    [0, 2]
+
+    Notes
+    -----
+    CTP is a generalization of the shortest path problem to graphs that are
+    partially observable.
+
+    The current solution is implemented using a repositioning algorithm. More
+    specifically: If f is the node from which we start, and t is the node where
+    we want to go, we perform the Dijkstra's algorithm to find the shortest path
+    from f in t. We begin to follow the path. If along the way, when we are
+    ready to cross link (u, v), we see that the link is closed, then we return
+    to our original f node and perform the Dijkstra's algorithm to find the
+    shortest path from f to t having removed from our link graph (u, v). We
+    begin to execute the new path found by f in t. If we see that a link we are
+    going to cross is closed, we go back to the beginning, we remove the
+    enclosed link of the graph, we execute the algorithm of Dijkstra, etc. So
+    again we perform the Dijkstra's algorithm as many times as we fall in a
+    closed link, but each time we return to the beginning. Our overall path
+    resulting to the feasible and appropriate path given the blocked edges.
+
+    The problem is said to have applications in operations research,
+    transportation planning, artificial intelligence, machine learning,
+    communication networks, and routing. A variant of the problem has been
+    studied for robot navigation with probabilistic landmark recognition
+
+    For more information, see here:
+    https://en.wikipedia.org/wiki/Canadian_traveller_problem
+
+    """
+
+    dis = nx.dijkstra_path(G, starting, ending)
+
+    if not nx.is_weighted(G):
+        raise nx.NetworkXError('Given graph is not weighted.')
+
+    for i in range(len(dis)-1):
+        dis_edge = dis[i], dis[i+1]
+        if dis_edge in blocked:
+            if dis_edge in G.edges():
+                G.remove_edge(dis[i], dis[i+1])
```

Comment:
Can you refactor this so that the algorithm doesn't modify the graph `G`, for example by using `G.copy()`?